nextcord/ext/commands/bot.py

```diff
@@ -1435,12 +1384,164 @@ class Bot(BotBase, nextcord.Client):
+            member_cache_flags=member_cache_flags,
+            chunk_guilds_at_startup=chunk_guilds_at_startup,
+            status=status,
+            activity=activity,
+            allowed_mentions=allowed_mentions,
+            heartbeat_timeout=heartbeat_timeout,
+            guild_ready_timeout=guild_ready_timeout,
+            assume_unsync_clock=assume_unsync_clock,
+            enable_debug_events=enable_debug_events,
+            loop=loop,
+            lazy_load_commands=lazy_load_commands,
+            rollout_associate_known=rollout_associate_known,
+            rollout_delete_unknown=rollout_delete_unknown,
+            rollout_register_new=rollout_register_new,
+            rollout_update_known=rollout_update_known,
+            rollout_all_guilds=rollout_all_guilds,
+        )
+
+        BotBase.__init__(
+            self,
+            command_prefix=command_prefix,  # type: ignore
+            help_command=help_command,
+            description=description,
+            owner_id=owner_id,
+            owner_ids=owner_ids,
+            strip_after_prefix=strip_after_prefix,
+            case_insensitive=case_insensitive,
+        )
 
 
 class AutoShardedBot(BotBase, nextcord.AutoShardedClient):
     """This is similar to :class:`.Bot` except that it is inherited from
     :class:`nextcord.AutoShardedClient` instead.
     """
 
-    pass
+    def __init__(
+        self,
+        command_prefix: Union[
+            _NonCallablePrefix,
+            Callable[[Self, Message], Union[Awaitable[_NonCallablePrefix], _NonCallablePrefix]],
+        ] = tuple(),
+        help_command: Optional[HelpCommand] = MISSING,
+        description: Optional[str] = None,
+        *,
+        max_messages: Optional[int] = 1000,
+        connector: Optional[aiohttp.BaseConnector] = None,
+        proxy: Optional[str] = None,
+        proxy_auth: Optional[aiohttp.BasicAuth] = None,
+        shard_id: Optional[int] = None,
+        shard_count: Optional[int] = None,
+        shard_ids: Optional[list[int]] = None,
+        application_id: Optional[int] = None,
+        intents: nextcord.Intents = nextcord.Intents.default(),
+        member_cache_flags: MemberCacheFlags = MISSING,
+        chunk_guilds_at_startup: bool = MISSING,
+        status: Optional[Status] = None,
+        activity: Optional[BaseActivity] = None,
+        allowed_mentions: Optional[AllowedMentions] = None,
+        heartbeat_timeout: float = 60.0,
+        guild_ready_timeout: float = 2.0,
+        assume_unsync_clock: bool = True,
+        enable_debug_events: bool = False,
+        loop: Optional[asyncio.AbstractEventLoop] = None,
+        lazy_load_commands: bool = True,
+        rollout_associate_known: bool = True,
+        rollout_delete_unknown: bool = True,
+        rollout_register_new: bool = True,
+        rollout_update_known: bool = True,
+        rollout_all_guilds: bool = False,
+        owner_id: Optional[int] = None,
+        owner_ids: Iterable[int] = set(),
```

Comment:
Using `set()` in a default is discouraged since it would be shared across multiple calls if modified. Not a huge problem in this context, but using `None` and setting it to `set()` within the function would be recommended.

https://rules.sonarsource.com/python/RSPEC-5717